### PR TITLE
CSI proxy integration for azurefile

### PR DESCRIPTION
**Pull request by kkmsft**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind test
> /kind failing-test

/kind feature

> /kind flake

**What this PR does / why we need it**:
Integrate with CSI proxy for azure file to work.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
**Release note**:
```
CSI proxy integration for windows
```

### CODE REVIEW

**Comment by andyzhangx** [comment]
pls run `hack/update-dependencies.sh` to fix go module issue:
```
Some files got changed after dependencies update
```

**Comment by andyzhangx** [comment]
@kkmsft is this PR completed? if so, pls change the title.

**Comment by andyzhangx** [comment]
cc @ZeroMagic for code review

**Comment by andyzhangx** [comment]
@kkmsft is there any remaining work items for this windows support? feel free to list to-do work items here, thanks.

File: pkg/mounter/safe_mounter_windows.go [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+
+	response, err := mounter.FsClient.IsMountPoint(context.Background(),
+		&fsv1alpha1.IsMountPointRequest{
+			Path: normalizeWindowsPath(path),
+		})
+	if err != nil {
+		klog.Error(err)
+		return false, err
+	}
+	klog.Error(response.IsMountPoint)
```
Comment (ZeroMagic) [inline pkg/mounter/safe_mounter_windows.go]: here are some the redundant log

File: pkg/mounter/safe_mounter_windows.go [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+	if err != nil {
+		klog.Error(err)
+		return false, err
+	}
+	if !isExists {
+		klog.Error("returning not exists")
+		return true, os.ErrNotExist
+	}
+
+	response, err := mounter.FsClient.IsMountPoint(context.Background(),
```
Comment (ZeroMagic) [inline pkg/mounter/safe_mounter_windows.go]: Do we need to handle the response.Error？Is it equal to the other return value `err`?

Comment (kkmsft) [reply pkg/mounter/safe_mounter_windows.go]: Initially the APIs had both error outside and error within. We decided to pass the errors only via the outside one. Will be removing those internal error in alpha2 version of the APIs.

Comment (ZeroMagic) [reply pkg/mounter/safe_mounter_windows.go]: Got it.

**Pull request merged**